sympy/solvers/solveset.py

```diff
@@ -246,10 +246,15 @@ def _invert_real(f, g_ys, symbol):
                 s, b = integer_log(rhs, base)
                 if b:
                     return _invert_real(expo, FiniteSet(s), symbol)
-            elif rhs is S.One:
-                # special case: 0**x - 1
-                return (expo, FiniteSet(0))
-            return (expo, S.EmptySet)
+                else:
+                    return _invert_real(expo, S.EmptySet, symbol)
+            elif base.is_zero:
+                one = Eq(rhs, 1)
+                if one == S.true:
+                    # special case: 0**x - 1
+                    return _invert_real(expo, FiniteSet(0), symbol)
+                elif one == S.false:
+                    return _invert_real(expo, S.EmptySet, symbol)
 
```

Comment:
Don't you need ` return (f, g_ys)` as the fallback return value if none of the above are true?